### DISCUSSION TITLE
Very small modification for Condensation

### DISCUSSION THREAD

**Discussion by Moss4t** [networkx/networkx#5570] [Algorithms]
I looked into the algorithm and found that adding one more input parameter 'offset' to the method, it can modify the start of the numbering.
The original: `def condensation(G, scc=None):`
It can be modified like so: `def condensation(G, scc=None, offset=0):`

Just need to add this offset to the `for i, component in enumerate(scc):` part of the code.
Like so: `for i, component in enumerate(scc, offset):`, and it will start the numbering of scc's from the offset.

This is a helpful small detail for me, and it would be nice to have.
Thank you in advance,
Moss4t

**Comment by rossbar** [comment]
If the desired feature is only to relabel the nodes in the condensed graph, then I think it would be better (and much more general) to handle the remapping explicitly instead of adding an extra kwarg to `condensation`. For example, to get behavior like `condensation(G, offset=3)`, you could instead do:

```
H = nx.condensation(G)
H = nx.relabel_nodes(H, mapping={n: n+3 for n in H.nodes})
```
It's slightly more typing, but it's also more general - i.e. the mapping can be whatever suits your usecase, rather than being bound to a numeric offset.

**Comment by Moss4t** [comment]
True. Thank you for the advice. I did not know about relabeling, that is
why I came up with this solution.

Ross Barnowski ***@***.***> ezt írta (időpont: 2022. ápr.
26., K de. 5:30):
